CMakeLists.txt

```diff
@@ -288,7 +288,7 @@ endif()
 hpx_option(HPX_WITH_DYNAMIC_HPX_MAIN BOOL
   "Enable dynamic overload of system ``main()`` (Linux only, default: ON)"
   ON ADVANCED)
-if(HPX_WITH_DYNAMIC_HPX_MAIN)
+if(HPX_WITH_DYNAMIC_HPX_MAIN AND (NOT MSVC))
```

Comment:
Should we also make it a hard configuration error if `HPX_WITH_DYNAMIC_HPX_MAIN=ON` on Windows? There might be quite a few others as well in which case we can do this separately?

Reply:
Let's do this separately. I agree we should turn non-applicable options into hard configuration errors.